src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonApiClient.kt

```diff
@@ -61,29 +47,18 @@ class PrisonApiClient(@Qualifier("oauthPrisonClient") val prisonerApiWebClient:
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
       .bodyToMono(typeReference<List<CourtEventOutcome>>())
-      .onErrorResume { webClientErrorHandler(it) }
       .block() ?: emptyList()
   }
 
-  private fun <API_RESPONSE_BODY_TYPE> webClientErrorHandler(exception: Throwable): Mono<API_RESPONSE_BODY_TYPE> =
+  private fun <API_RESPONSE_BODY_TYPE> coerce404ResponseToNull(exception: Throwable): Mono<API_RESPONSE_BODY_TYPE> =
     with(exception) {
-      if (this is WebClientResponseException) {
-        val uriPath = request?.uri?.path
-        when (statusCode) {
-          FORBIDDEN -> {
-            log.error("Client token does not have correct role to call prisoner-api $uriPath")
-          }
-
-          NOT_FOUND -> {
-            log.info("No resource found when calling prisoner-api $uriPath")
-          }
-
-          else -> {
-            log.error("Failed to call prisoner-api $uriPath [statusCode: $statusCode, body: ${this.responseBodyAsString}]")
```

Comment:
This wasn't logging an exception as a separate parameter meaning it wouldn't be recorded as an exception in app insights